python/lsst/ts/m2/csc.py

```diff
@@ -684,27 +689,32 @@ def _assert_enabled_csc_and_controller(
         self.controller_cell.assert_controller_state(message_name, allowed_curr_states)
         self.assert_enabled()
 
-    async def _write_command_to_server(
-        self, message_name: str, timeout: float, message_details: dict | None = None
+    async def _execute_command_to_server(
+        self,
+        command: typing.Coroutine,
+        *args: typing.Any,
+        timeout: float | None = None,
+        **kwargs: dict[str, typing.Any],
     ) -> None:
-        """Write the command to server.
+        """Execute the command to server.
 
         Parameters
         ----------
-        message_name : `str`
-            Message name to server.
-        timeout : `float`
-            Timeout of command in second.
-        message_details : `dict` or None, optional
-            Message details. (the default is None)
+        command : `coroutine`
+            Command.
+        *args : `args`
+            Arguments of the command.
+        timeout : `float` or None, optional
+            Timeout of command in second. If None, the default value is used.
+            (the default is None)
+        **kwargs : `dict`, optional
+            Additional keyword arguments to run the command.
         """
 
+        timeout = self.COMMAND_TIMEOUT if (timeout is None) else timeout
+
         try:
-            await self.controller_cell.write_command_to_server(
-                message_name,
-                message_details=message_details,
-                timeout=timeout,
-            )
+            await command(*args, timeout=timeout, **kwargs)  # type: ignore[operator]
 
         except OSError:
```

Comment:
I know this is not part of the PR but maybe document in the docstring that the method captures `OSError` and sends the CSC to fault. Also, consider logging the exception before sending the CSC to fault, e.g.;

```
except OSError:
    self.log.exeception("Connection error executing command.")
    await self.controller_cell.close()
    await self.fault(...)
```

Reply:
Updated in the commit: [71466d9](https://github.com/lsst-ts/ts_m2/pull/45/commits/71466d9802e0a0c6b2ae58e37845e0cd3430fc94). Thanks!